uat/testing-features/src/main/java/com/aws/greengrass/steps/CloudWatchSteps.java

```diff
@@ -86,13 +111,22 @@ public void verifyCloudWatchGroupWithStreams(String componentType, String compon
     }
 
     private boolean doesStreamExistInGroup(String logGroupName, String streamName) {
-        List<LogStream> streams = logsLifecycle.findStream(logGroupName, streamName);
-        boolean exists = streams.stream().anyMatch(stream -> stream.logStreamName().matches(streamName));
+        try {
+            // The OTF watch steps check evey 100ms this to avoids hammering the api. Ideally OTF
+            // can allow us to configure the check interval rate
+            Thread.sleep(VERIFICATION_RATE_MILLISECONDS);
+            List<LogStream> streams = logsLifecycle.findStream(logGroupName, streamName);
+            boolean exists = streams.stream().anyMatch(stream -> stream.logStreamName().matches(streamName));
```

Comment:
What's the difference between these two?

Reply:
Between which two? Well technically findStream gets all the streams that match a prefix it works using a fuzzy search so if there are 2 streams.